frontend/public/components/modals/toleration-update-modal.tsx

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as React from 'react';
+import { PromiseComponent, Dropdown } from '../utils';
+import { K8sKind, k8sPatch } from '../../module/k8s';
+
+import { createModalLauncher, ModalBody, ModalComponentProps, ModalSubmitFooter, ModalTitle } from '../factory';
+import * as _ from 'lodash-es';
+
+class TolerationUpdateModal extends PromiseComponent {
+  readonly state: TolerationUpdateModalState;
+
+  constructor(public props: TolerationUpdateModalProps) {
+    super(props);
+    this.state.tolerations = this.props.resource.spec.tolerations;
```

Comment:
This path will need change change depending on whether it's a pod or other workload.